Update index.md (#196)
JuliaImages/ImageCore.jl: docs/src/index.md

useful low-level functions (including "traits") that simplify image
display, input/output, and the writing of algorithms.

Some of the key features and functionalities provided by ImageCore.jl include:

1.Image data representation: It provides the Image type to store image data along with functions for basic image manipulation.

> 2.Image rotation and flipping: Functions to rotate and flip images.

Reviewer: ```suggestion
- Storage type transformations: ImageCore.jl provides various methods like `float32`, `n0f8`, etc which can be used to transform raw storage type of the image without changing the color space. 
- Image view conversion: It provides various methods like `rawview`, `channelview`, `colorview` , `stackedview` which help in conversions between various types of representations. Default representation of images is likely all that is needed most of the time but these functions provided more flexibility while working with image data.
- Pixel level transformations: Several methods like `clamp01` , `clamp01nan`, `scaleminmax`, etc are available that help with truncating image pixel values within certain valid limits and helps to avoid IO errors.
- Image Traits: These can prove to be useful to users when trying to add more meaning to different axes of image and to add dimension specific information.
```

It is important to note that ImageCore.jl focuses on fundamental image processing operations and may not have the full range of advanced image processing capabilities found in more specialized libraries. For more complex image processing tasks, you might need to explore other Julia packages such as ImageFiltering.jl, or other external libraries.

If you want to use ImageCore.jl, you can add it to your Julia environment using the package manager. Open a Julia REPL and use the following command:

using Pkg

> Pkg.add("ImageCore")

Reviewer: ``````suggestion
```jl
] add ImageCore
```
``````

useful low-level functions (including "traits") that simplify image
display, input/output, and the writing of algorithms.

Some of the key features and functionalities provided by ImageCore.jl include:

> 1.Image I/O: Loading and saving images in various formats like PNG, JPEG, BMP, TIFF, etc.

Reviewer: Image IO support is provided by FileIO.jl and not by this package

2.Image data representation: It provides the Image type to store image data along with functions for basic image manipulation.
3.Resizing and scaling: Functions for resizing and scaling images.

> 4.Image rotation and flipping: Functions to rotate and flip images.

Reviewer: Resizing, scaling, image rotation and flipping are provided by ImageTransformations.jl and exported there

> 5.Basic image processing operations: Some simple image processing operations like converting images to grayscale, computing histograms, etc.

Reviewer: histograms computation is not provided here, need to update it

> It is important to note that ImageCore.jl focuses on fundamental image processing operations and may not have the full range of advanced image processing capabilities found in more specialized libraries. For more complex image processing tasks, you might need to explore other Julia packages such as ImageMagick.jl or ImageFiltering.jl, or other external libraries.

Reviewer: ImageMagick.jl provides just IO methods, you can see different packages and their use cases here: https://juliaimages.org/latest/pkgs/

useful low-level functions (including "traits") that simplify image
display, input/output, and the writing of algorithms.

Some of the key features and functionalities provided by ImageCore.jl include:

- Storage type transformations: ImageCore.jl provides various methods like `float32`, `n0f8`, etc which can be used to transform raw storage type of the image without changing the color space. 

> - Image view conversion: It provides various methods like `rawview`, `channelview`, `colorview` , `stackedview` which help in conversions between various types of representations. Default representation of images is likely all that is needed most of the time but these functions provided more flexibility while working with image data.

Reviewer: ```suggestion
- Image view conversion: It provides various methods like `rawview`, `channelview`, `colorview` , `stackedview` which help in conversion between various types of representations. Default representation of images is likely all that is needed most of the time but these functions provided more flexibility while working with image data.
```

- Pixel level transformations: Several methods like `clamp01` , `clamp01nan`, `scaleminmax`, etc are available that help with truncating image pixel values within certain valid limits and helps to avoid IO errors.
- Image Traits: These can prove to be useful to users when trying to add more meaning to different axes of image and to add dimension specific information.

> 

Reviewer: ```suggestion
```

> ImageCore often acts as the "common core" dependency for working with any kind of image in the JuliaImages ecosystem., but itself provides almost none of the standard image-processing algorithms. These algorithms are found in other packages in the JuliaImages ecosystem.

Reviewer 2: ```suggestion
ImageCore often acts as the "common core" dependency for working with any kind of image in the JuliaImages ecosystem, but itself provides almost none of the standard image-processing algorithms. These algorithms are found in other packages in the JuliaImages ecosystem.
```
oops, sorry about that

> It is important to note that ImageCore.jl focuses on fundamental image processing operations and may not have the full range of advanced image processing capabilities found in more specialized libraries. For more complex image processing tasks, you might need to explore other Julia packages such as ImageFiltering.jl, or other external libraries.

Reviewer: ```suggestion
It is important to note that ImageCore.jl focuses on fundamental image processing operations and may not have the full range of advanced image processing capabilities found in more specialized libraries. For more complex image processing tasks, you might want to explore other  JuliaImages packages such as ImageFiltering.jl, ImageTransformations, etc.
```

> It is important to note that ImageCore.jl focuses on fundamental image processing operations and may not have the full range of advanced image processing capabilities found in more specialized libraries. For more complex image processing tasks, you might need to explore other Julia packages such as ImageFiltering.jl, or other external libraries.

Reviewer 2: ```suggestion
ImageCore often acts as the "common core" dependency for working with any kind of image in the JuliaImages ecosystem., but itself provides almost none of the standard image-processing algorithms. These algorithms are found in other packages in the JuliaImages ecosystem.
```

> If you want to use ImageCore.jl, you can add it to your Julia environment using the package manager. Open a Julia REPL and use the following command:

Reviewer 2: ```suggestion
If you want to use ImageCore.jl, you can add it to your project environment using the package manager. Open a Julia REPL and use the following command:
```
"Julia environment" to me implies your default environment, but I keep only developer and plotting tools in my default environment. Packages like ImageCore get added to specific project environments.

```jl
] add ImageCore
```

> 

Reviewer: ```suggestion
```
